mila/task_4/main.py

```diff
@@ -0,0 +1,72 @@
+import random
+import urllib.request
+
+ALLOWED_CHARACTERS = 'abcdefghijklmnopqrstuvwxyz '
+
+
+def analyze_text(url):
+    with urllib.request.urlopen(url) as response:
+        text = response.read().decode('utf-8').lower()
+
+        global count
+        count = 0
+
+        hashmap = {char: {letter: 0 for letter in ALLOWED_CHARACTERS} for char in ALLOWED_CHARACTERS}
+
+        for i, char in enumerate(text):
+            if char in ALLOWED_CHARACTERS:
+                next_index = i + 1
+                count += 1
+
+                while next_index < len(text) and not text[next_index] in ALLOWED_CHARACTERS:
+                    next_index += 1
+
+                if next_index < len(text):
+                    next_char = text[next_index]
+                    hashmap[char][next_char] += 1
+
+        return hashmap
+
+
+def display(hashmap):
+    user_input = input("Enter a letter: ").strip().lower()
+
+    if len(user_input) == 1 and user_input in ALLOWED_CHARACTERS:
+        return generate_text(hashmap, user_input, 200)
+    else:
+        return 'Invalid value!'
+
+
+def generate_text(hashmap, input, given_length):
+    generated = input
+    current_char = input
+
+    for char in hashmap:
+        # sum of the values for the current char's hashmap
+        total_sum = sum(hashmap[char].values())
+
+        # supposed sum of the values for the current char's new scaled hashmap
+        scaled_sum = round(total_sum / count) * given_length
+
+        for next_char in hashmap[char]:
+            if total_sum and scaled_sum:
+                hashmap[char][next_char] = round((hashmap[char][next_char] / total_sum) * scaled_sum)
+
+    while len(generated) != given_length:
+        next_probabilities = hashmap[current_char]
```

Comment:
It's probably can be a chance that current_char key doesn't exist in hashmap

Reply:
"current_char" is obtained from the user's input. In the display() function, we validate user input by checking if the character is allowed, which guarantees it exists in the hashmap 👍